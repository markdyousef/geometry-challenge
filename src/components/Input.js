import React from "react";
import styled from "styled-components";
import Label from "./Label";

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 5px;
`;

const Input = styled.input`
  padding: 10px 8px;
  width: 40px;
  border-radius: 3px;
  border: 1px solid #ddd;
  font-size: 15px;
  outline: none;
  text-align: center;
  margin: 5px;
`;

export default class extends React.PureComponent {
  render() {
    return (
      <InputContainer>
        <Label name={this.props.id} />
        <Input
          id={this.props.id}
          value={this.props.value}
          onChange={this.props.handleChange}
          type="number"
          minLength={1}
          maxLength={1000}
        />
      </InputContainer>
    );
  }
}
