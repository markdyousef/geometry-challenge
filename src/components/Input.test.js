import React from "react";
import { shallow } from "enzyme";
import Input from "./Input";

it("renders Input", () => {
  const wrapper = shallow(<Input id="a" value={5} onChange={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
