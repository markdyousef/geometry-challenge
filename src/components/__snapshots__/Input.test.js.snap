// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    id="a"
    onChange={[Function]}
    value={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <PureComponent
          name="a"
/>,
        <styled.input
          id="a"
          maxLength={1000}
          minLength={1}
          onChange={undefined}
          type="number"
          value={5}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "a",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "a",
          "maxLength": 1000,
          "minLength": 1,
          "onChange": undefined,
          "type": "number",
          "value": 5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PureComponent
            name="a"
/>,
          <styled.input
            id="a"
            maxLength={1000}
            minLength={1}
            onChange={undefined}
            type="number"
            value={5}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "a",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "a",
            "maxLength": 1000,
            "minLength": 1,
            "onChange": undefined,
            "type": "number",
            "value": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
